public/docs/ts/latest/cookbook/aot-compiler.jade

```diff
@@ -301,3 +302,130 @@ a#source-code
+  Our source code is for the most part agnostic of how we compile the application, but there are a few differences to keep in mind.
+
+  *Index.html*
+
+  The initial loading is different between JiT and AoT applications. In our case we have decided to use two different index.html files. 
+
+  Our JiT version relies on `SystemJS` for loading of individual modules, but our AoT version loads the entire application as a single script tag without a dependency on `SystemJS`. This means we can remove the `SystemJS` script tag when doing AoT. It turns out we can also remove the dependency on `reflect-metadata`.  
+
+  We have included the two index.html files below:
+
++makeTabs(
+  `toh-6/ts/index.html,
+   toh-6-aot/ts/index.html`,
+  null,
+  `index.html (JiT),
+   index.html (AoT)`
+)
+
+:marked
+  *External file paths*
+
+  The AoT compiler assumes component-relative paths when locating external templates and css files. This differs from JiT where the path is specified as either an absolute path from the application root, or a component-relative path in tandem with a `module.id`.  
+
+  To make our components compatible with both AoT and JiT we have added `module.id` and component-relative paths to all our components.
+
+  The AoT compiler will ignore `module.id` during compilation, but there is one caveat. `module.id` is included in the final JavaScript, but in the case of AoT, `module` will not be defined at runtime. This means we will get a runtime error when trying to access `id` on the undefined `module` object. 
+
+  The solution is simple. Since `module.id` has no meaning in an AoT application we can work around this by assigning `window.module` an arbitrary value at runtime.
+
+  The assigned value can be anything. The important part is to define `window.module` to avoid failure when `module.id` is referenced.
+
++makeExample('toh-6-aot/ts/index.html','moduleId','index.html (module.id)')(format='.')  
+
+:marked
+  Manually setting `module` to an arbitrary value may seem a bit hacky, but we expect this to be a temporary workaround. 
+
+:marked
+  *TypeScript configuration*
+
+  To prepare our code for Tree Shaking our AoT compiled application is targeting `ES2015` modules. This is not ideal for our JiT configuration since most browsers can't execute `ES2015` directly.
+
+  In order to output regular `ES5` JavaScript for out JiT setup, we have created a second `tsconfig.json` file.
+
+  The two TypeScript configuration files can be seen below:
+
++makeTabs(
+  `toh-6/ts/tsconfig.json,
+   toh-6/ts/tsconfig-aot.json`,
+  null,
+  `tsconfig.json (JiT),
+   tsconfig-aot.json (AoT)`
+)   
+
+:marked
+  ### External AoT Libraries
+
+  Tour of Heroes has an external dependency on the `angular-in-memory-web-api` library.
+
+  We can't expect all libraries to be released with AoT in mind, but `angular-in-memory-web-api` has already been AoT compiled and released with the necessary dependencies to integrate it with another AoT compiled application.
```

Comment:
I think we should expect this since one non-AoT provided library prevents AoT altogether. Maybe that is addressed in a separate cookbook on creating a third party library.